x-pack/plugins/alerting/server/routes/get_global_execution_logs.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import { ILicenseState } from '../lib';
+import { GetGlobalExecutionLogParams } from '../rules_client';
+import { RewriteRequestCase, verifyAccessAndContext } from './lib';
+import { AlertingRequestHandlerContext, INTERNAL_BASE_ALERTING_API_PATH } from '../types';
+
+const sortOrderSchema = schema.oneOf([schema.literal('asc'), schema.literal('desc')]);
+
+const sortFieldSchema = schema.oneOf([
+  schema.object({ timestamp: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ execution_duration: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ total_search_duration: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ es_search_duration: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ schedule_delay: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ num_triggered_actions: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ num_generated_actions: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ num_active_alerts: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ num_recovered_alerts: schema.object({ order: sortOrderSchema }) }),
+  schema.object({ num_new_alerts: schema.object({ order: sortOrderSchema }) }),
+]);
+
+const sortFieldsSchema = schema.arrayOf(sortFieldSchema, {
+  defaultValue: [{ timestamp: { order: 'desc' } }],
+});
+
+const querySchema = schema.object({
+  date_start: schema.string(),
+  date_end: schema.maybe(schema.string()),
+  filter: schema.maybe(schema.string()),
+  per_page: schema.number({ defaultValue: 10, min: 1 }),
+  page: schema.number({ defaultValue: 1, min: 1 }),
+  sort: sortFieldsSchema,
+});
+
+const rewriteReq: RewriteRequestCase<GetGlobalExecutionLogParams> = ({
+  date_start: dateStart,
+  date_end: dateEnd,
+  per_page: perPage,
+  ...rest
+}) => ({
+  ...rest,
+  dateStart,
+  dateEnd,
+  perPage,
+});
+
+export const getGlobalExecutionLogRoute = (
```

Comment:
nit: could we define this inside `x-pack/plugins/alerting/server/routes/get_rule_execution_log.ts`? Separate route just defined in the same file so we can reuse the query and sort schemas? Or alternatively extract out the schemas into a separate file that's used by both route definitions?

Reply:
So when I try to combine these files and define:

```js
const rewriteReq: RewriteRequestCase<GetGlobalExecutionLogParams | GetExecutionLogByIdParams>
```

Typescript complains that the `id` property is missing from `GetGlobalExecutionLogParams` but required in `GetExecutionLogByIdParams`. Meanwhile the actual functionality of `rewriteReq` doesn't include `id` at all and is the same for both of them, so duplicating this function just for the second type signature would be even weirder than separating them into two files.

I know it's a lot of repeated code but I think it's similar to follow the convention of one file per route in this case.